pkg/engine/mutate/patchesUtils.go

```diff
@@ -34,6 +33,127 @@ func generatePatches(src, dst []byte) ([][]byte, error) {
 	return patchesBytes, err
 }
 
+// filtersAndSortsPatches
+// 1. filters out patches with the certain paths
+// 2. sorts the removal patches(with same path) by the key of index
+// in descending order. For example, when removes 2 elements from an
+// array, once the first is removed, index 1 will be invalid
+//
```

Comment:
Change, "For example..." to 

"The sort is required as when removing multiple elements from an array, the elements must be removed in descending order to preserve each index value."